core/src/main/java/com/linecorp/armeria/server/logging/LoggingServiceBuilder.java

```diff
@@ -0,0 +1,191 @@
+     * {@link Function} remove sensitive headers, like {@code Cookie}, before logging. If unset, will use
+     * {@link Function#identity()}.
+     */
+    public LoggingServiceBuilder requestHeadersSanitizer(
+            Function<HttpHeaders, HttpHeaders> requestHeadersSanitizer) {
+        this.requestHeadersSanitizer = requireNonNull(requestHeadersSanitizer, "requestHeadersSanitizer");
+        return this;
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize request content before logging. It is common to have the
+     * {@link Function} remove sensitive content, such as an GPS location query, before logging. If unset,
+     * will use {@link Function#identity()}.
+     */
+    public LoggingServiceBuilder requestContentSanitizer(Function<Object, Object> requestContentSanitizer) {
+        this.requestContentSanitizer = requireNonNull(requestContentSanitizer, "requestContentSanitizer");
+        return this;
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize response headers before logging. It is common to have the
+     * {@link Function} remove sensitive headers, like {@code Set-Cookie}, before logging. If unset,
+     * will use {@link Function#identity()}.
+     */
+    public LoggingServiceBuilder responseHeadersSanitizer(
+            Function<HttpHeaders, HttpHeaders> responseHeadersSanitizer) {
+        this.responseHeadersSanitizer = requireNonNull(responseHeadersSanitizer, "responseHeadersSanitizer");
+        return this;
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize response content before logging. It is common to have the
+     * {@link Function} remove sensitive content, such as an address, before logging. If unset,
+     * will use {@link Function#identity()}.
+     */
+    public LoggingServiceBuilder responseContentSanitizer(Function<Object, Object> responseContentSanitizer) {
+        this.responseContentSanitizer = requireNonNull(responseContentSanitizer, "responseContentSanitizer");
+        return this;
+    }
+
+    /**
+     * Sets the rate at which to sample requests to log. Any number between {@code 0.0} and {@code 1.0} will
+     * cause a random sample of the requests to be logged. The random sampling is appropriate for low-traffic
+     * (ex servers that each receive &lt;100K requests). If unset, all requests will be logged.
+     */
+    public LoggingServiceBuilder samplingRate(float samplingRate) {
+        checkArgument(0.0 <= samplingRate && samplingRate <= 1.0, "samplingRate must be between 0.0 and 1.0");
+        this.sampler = Sampler.create(samplingRate);
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link LoggingService} decorating {@code delegate} based on the properties of
+     * this builder.
+     */
+    public <I extends Request, O extends Response> LoggingService<I, O> build(Service<I, O> delegate) {
```

Comment:
@trustin I'm considering in this PR to move this to `common/logging/LoggingDecoratorBuilder`, rename to `buildService` and `newServiceDecorator` since the options will be identical for service and client. And then can add `buildClient` and `newClientDecorator` in a following PR. What do you think?